syntax = "proto3";

service DFS {
    rpc StoreChunk (StoreChunkRequest) returns (StoreChunkResponse);
    rpc RetrieveChunk (RetrieveChunkRequest) returns (RetrieveChunkResponse);
}

message StoreChunkRequest {
    string file_name = 1;
    int32 chunk_index = 2;
    bytes chunk_data = 3;
}

message StoreChunkResponse {
    string status = 1;
}

message RetrieveChunkRequest {
    string file_name = 1;
    int32 chunk_index = 2;
}

message RetrieveChunkResponse {
    bytes chunk_data = 1;
}